shared/wallets/wallet/settings/container.js

```diff
@@ -16,7 +16,12 @@ const mapStateToProps = (state, {routeProps}) => {
   const user = account.isDefault ? me : ''
   const currencies = Constants.getDisplayCurrencies(state)
   const currency = Constants.getDisplayCurrency(state, accountID)
-  const currencyWaiting = anyWaiting(state, Constants.changeDisplayCurrencyWaitingKey)
+  const currencyWaiting = anyWaiting(
+    state,
+    Constants.changeDisplayCurrencyWaitingKey,
+    Constants.getDisplayCurrencyWaitingKey(accountID)
+  )
+  const saveCurrencyWaiting = anyWaiting(state, Constants.changeDisplayCurrencyWaitingKey)
```

Comment:
This needs a separate flag so we don't render `SaveIndicator` on navigating into the settings page on desktop (because we reload the display currency every time)